diff --git a/rules/aip0142/BUILD.bazel b/rules/aip0142/BUILD.bazel
new file mode 100644
index 0000000..726e4ae
--- /dev/null
+++ b/rules/aip0142/BUILD.bazel
@@ -0,0 +1,33 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "aip0142",
+    srcs = [
+        "aip0142.go",
+        "time_field_names.go",
+        "time_field_type.go",
+    ],
+    importpath = "github.com/googleapis/api-linter/rules/aip0142",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//lint",
+        "//locations",
+        "//rules/internal/utils",
+        "@com_github_jhump_protoreflect//desc",
+        "@org_bitbucket_creachadair_stringset//:stringset",
+    ],
+)
+
+go_test(
+    name = "aip0142_test",
+    srcs = [
+        "aip0142_test.go",
+        "time_field_names_test.go",
+        "time_field_type_test.go",
+    ],
+    embed = [":aip0142"],
+    deps = [
+        "//lint",
+        "//rules/internal/testutils",
+    ],
+)
