diff --git a/rules/aip0158/BUILD.bazel b/rules/aip0158/BUILD.bazel
new file mode 100644
index 0000000..358a7f4
--- /dev/null
+++ b/rules/aip0158/BUILD.bazel
@@ -0,0 +1,46 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "aip0158",
+    srcs = [
+        "aip0158.go",
+        "request_page_size_field.go",
+        "request_page_token_field.go",
+        "request_skip_field.go",
+        "response_next_page_token_field.go",
+        "response_plural_first_field.go",
+        "response_repeated_first_field.go",
+        "response_unary.go",
+    ],
+    importpath = "github.com/googleapis/api-linter/rules/aip0158",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//lint",
+        "//locations",
+        "//rules/internal/utils",
+        "@com_github_jhump_protoreflect//desc",
+        "@com_github_jhump_protoreflect//desc/builder",
+        "@com_github_stoewer_go_strcase//:go-strcase",
+    ],
+)
+
+go_test(
+    name = "aip0158_test",
+    srcs = [
+        "aip0158_test.go",
+        "request_page_size_field_test.go",
+        "request_page_token_field_test.go",
+        "request_skip_field_test.go",
+        "response_next_page_token_field_test.go",
+        "response_plural_first_field_test.go",
+        "response_repeated_first_field_test.go",
+        "response_unary_test.go",
+    ],
+    embed = [":aip0158"],
+    deps = [
+        "//lint",
+        "//rules/internal/testutils",
+        "@com_github_jhump_protoreflect//desc",
+        "@com_github_jhump_protoreflect//desc/builder",
+    ],
+)
