diff --git a/rules/internal/testutils/BUILD.bazel b/rules/internal/testutils/BUILD.bazel
new file mode 100644
index 0000000..df22700
--- /dev/null
+++ b/rules/internal/testutils/BUILD.bazel
@@ -0,0 +1,37 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "testutils",
+    srcs = [
+        "parse.go",
+        "problems.go",
+    ],
+    importpath = "github.com/googleapis/api-linter/rules/internal/testutils",
+    visibility = ["//rules:__subpackages__"],
+    deps = [
+        "//lint",
+        "@com_github_google_go_cmp//cmp",
+        "@com_github_jhump_protoreflect//desc",
+        "@com_github_jhump_protoreflect//desc/protoparse",
+        "@com_github_lithammer_dedent//:dedent",
+        "@com_google_cloud_go_longrunning//autogen/longrunningpb",
+        "@org_golang_google_genproto//googleapis/type/date",
+        "@org_golang_google_genproto//googleapis/type/datetime",
+        "@org_golang_google_genproto//googleapis/type/timeofday",
+        "@org_golang_google_genproto_googleapis_api//annotations",
+    ],
+)
+
+go_test(
+    name = "testutils_test",
+    srcs = [
+        "parse_test.go",
+        "problems_test.go",
+    ],
+    embed = [":testutils"],
+    deps = [
+        "//lint",
+        "@com_github_jhump_protoreflect//desc",
+        "@com_github_jhump_protoreflect//desc/builder",
+    ],
+)
