diff --git a/rules/aip0127/BUILD.bazel b/rules/aip0127/BUILD.bazel
new file mode 100644
index 0000000..51ea123
--- /dev/null
+++ b/rules/aip0127/BUILD.bazel
@@ -0,0 +1,38 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "aip0127",
+    srcs = [
+        "aip0127.go",
+        "http_annotation.go",
+        "http_template_pattern.go",
+        "http_template_syntax.go",
+        "resource_name_extraction.go",
+        "uri_leading_slash.go",
+    ],
+    importpath = "github.com/googleapis/api-linter/rules/aip0127",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//lint",
+        "//locations",
+        "//rules/internal/utils",
+        "@com_github_jhump_protoreflect//desc",
+    ],
+)
+
+go_test(
+    name = "aip0127_test",
+    srcs = [
+        "aip0127_test.go",
+        "http_annotation_test.go",
+        "http_template_pattern_test.go",
+        "http_template_syntax_test.go",
+        "resource_name_extraction_test.go",
+        "uri_leading_slash_test.go",
+    ],
+    embed = [":aip0127"],
+    deps = [
+        "//lint",
+        "//rules/internal/testutils",
+    ],
+)
