diff --git a/rules/aip0133/BUILD.bazel b/rules/aip0133/BUILD.bazel
new file mode 100644
index 0000000..d2cc264
--- /dev/null
+++ b/rules/aip0133/BUILD.bazel
@@ -0,0 +1,71 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "aip0133",
+    srcs = [
+        "aip0133.go",
+        "http_body.go",
+        "http_method.go",
+        "http_uri_parent.go",
+        "http_uri_resource.go",
+        "method_signature.go",
+        "request_id_field.go",
+        "request_message_name.go",
+        "request_parent_behavior.go",
+        "request_parent_field.go",
+        "request_parent_reference.go",
+        "request_parent_required.go",
+        "request_required_fields.go",
+        "request_resource_behavior.go",
+        "request_resource_field.go",
+        "request_unknown_fields.go",
+        "resource_reference_type.go",
+        "response_lro.go",
+        "response_message_name.go",
+        "synonyms.go",
+    ],
+    importpath = "github.com/googleapis/api-linter/rules/aip0133",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//lint",
+        "//locations",
+        "//rules/internal/utils",
+        "@com_github_jhump_protoreflect//desc",
+        "@com_github_jhump_protoreflect//desc/builder",
+        "@com_github_stoewer_go_strcase//:go-strcase",
+        "@org_bitbucket_creachadair_stringset//:stringset",
+    ],
+)
+
+go_test(
+    name = "aip0133_test",
+    srcs = [
+        "aip0133_test.go",
+        "http_body_test.go",
+        "http_method_test.go",
+        "http_uri_parent_test.go",
+        "http_uri_resource_test.go",
+        "method_signature_test.go",
+        "request_id_field_test.go",
+        "request_message_name_test.go",
+        "request_parent_behavior_test.go",
+        "request_parent_field_test.go",
+        "request_parent_reference_test.go",
+        "request_parent_required_test.go",
+        "request_required_fields_test.go",
+        "request_resource_behavior_test.go",
+        "request_resource_field_test.go",
+        "request_unknown_fields_test.go",
+        "resource_reference_type_test.go",
+        "response_lro_test.go",
+        "response_message_name_test.go",
+        "synonyms_test.go",
+    ],
+    embed = [":aip0133"],
+    deps = [
+        "//lint",
+        "//rules/internal/testutils",
+        "@com_github_jhump_protoreflect//desc",
+        "@com_github_jhump_protoreflect//desc/builder",
+    ],
+)
