diff --git a/rules/aip0148/BUILD.bazel b/rules/aip0148/BUILD.bazel
new file mode 100644
index 0000000..a8e282f
--- /dev/null
+++ b/rules/aip0148/BUILD.bazel
@@ -0,0 +1,44 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "aip0148",
+    srcs = [
+        "aip0148.go",
+        "declarative_friendly_fields.go",
+        "field_behavior.go",
+        "human_names.go",
+        "ip_address_format.go",
+        "uid_format.go",
+        "use_uid.go",
+    ],
+    importpath = "github.com/googleapis/api-linter/rules/aip0148",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//lint",
+        "//locations",
+        "//rules/internal/utils",
+        "@com_github_jhump_protoreflect//desc",
+        "@org_bitbucket_creachadair_stringset//:stringset",
+        "@org_golang_google_genproto_googleapis_api//annotations",
+        "@org_golang_google_protobuf//types/descriptorpb",
+    ],
+)
+
+go_test(
+    name = "aip0148_test",
+    srcs = [
+        "aip0148_test.go",
+        "declarative_friendly_fields_test.go",
+        "field_behavior_test.go",
+        "human_names_test.go",
+        "ip_address_format_test.go",
+        "uid_format_test.go",
+        "use_uid_test.go",
+    ],
+    embed = [":aip0148"],
+    deps = [
+        "//lint",
+        "//rules/internal/testutils",
+        "@org_bitbucket_creachadair_stringset//:stringset",
+    ],
+)
