diff --git a/rules/aip0134/BUILD.bazel b/rules/aip0134/BUILD.bazel
new file mode 100644
index 0000000..e017d0a
--- /dev/null
+++ b/rules/aip0134/BUILD.bazel
@@ -0,0 +1,63 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "aip0134",
+    srcs = [
+        "aip0134.go",
+        "http_body.go",
+        "http_method.go",
+        "http_uri_name.go",
+        "method_signature.go",
+        "request_allow_missing_field.go",
+        "request_mask_field.go",
+        "request_mask_required.go",
+        "request_message_name.go",
+        "request_required_fields.go",
+        "request_resource_field.go",
+        "request_resource_required.go",
+        "request_unknown_fields.go",
+        "response_lro.go",
+        "response_message_name.go",
+        "synonyms.go",
+    ],
+    importpath = "github.com/googleapis/api-linter/rules/aip0134",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//lint",
+        "//locations",
+        "//rules/internal/utils",
+        "@com_github_jhump_protoreflect//desc",
+        "@com_github_stoewer_go_strcase//:go-strcase",
+        "@org_bitbucket_creachadair_stringset//:stringset",
+    ],
+)
+
+go_test(
+    name = "aip0134_test",
+    srcs = [
+        "aip0134_test.go",
+        "http_body_test.go",
+        "http_method_test.go",
+        "http_uri_name_test.go",
+        "method_signature_test.go",
+        "request_allow_missing_field_test.go",
+        "request_mask_field_test.go",
+        "request_mask_required_test.go",
+        "request_message_name_test.go",
+        "request_required_fields_test.go",
+        "request_resource_field_test.go",
+        "request_resource_required_test.go",
+        "request_unknown_fields_test.go",
+        "response_lro_test.go",
+        "response_message_name_test.go",
+        "synonyms_test.go",
+    ],
+    embed = [":aip0134"],
+    deps = [
+        "//lint",
+        "//rules/internal/testutils",
+        "@com_github_jhump_protoreflect//desc",
+        "@com_github_jhump_protoreflect//desc/builder",
+        "@org_golang_google_genproto//protobuf/field_mask",
+    ],
+)
