diff --git a/rules/aip0163/BUILD.bazel b/rules/aip0163/BUILD.bazel
new file mode 100644
index 0000000..df55ada
--- /dev/null
+++ b/rules/aip0163/BUILD.bazel
@@ -0,0 +1,32 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "aip0163",
+    srcs = [
+        "aip0163.go",
+        "declarative_friendly_required.go",
+        "synonyms.go",
+    ],
+    importpath = "github.com/googleapis/api-linter/rules/aip0163",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//lint",
+        "//locations",
+        "//rules/internal/utils",
+        "@com_github_jhump_protoreflect//desc",
+    ],
+)
+
+go_test(
+    name = "aip0163_test",
+    srcs = [
+        "aip0163_test.go",
+        "declarative_friendly_required_test.go",
+        "synonyms_test.go",
+    ],
+    embed = [":aip0163"],
+    deps = [
+        "//lint",
+        "//rules/internal/testutils",
+    ],
+)
